### PR TITLE
send velocity setpoint to enable proper takeoff/land

### PR DESCRIPTION
**Pull request by mrivi**
Send velocity and position setpoints from AutoMapper to enable proper landing. Before this PR, the avoidance only knew about the position waypoint but due to the drift in the local frame that waypoint could have been reached while still being in air. 

Needs MAVROS PR https://github.com/mavlink/mavros/pull/1233 otherwise it isn't possible to send to the companion computer vectors containing NAN.

### CODE REVIEW

File: src/lib/FlightTasks/tasks/AutoMapper/FlightTaskAutoMapper.cpp [hunk 1]
```diff
@@ -86,6 +86,8 @@ bool FlightTaskAutoMapper::update()
 		_generateVelocitySetpoints();
 	}
 
+	_obstacle_avoidance.updateAvoidanceDesiredSetpoints(_position_setpoint, _velocity_setpoint);
```
Comment (MaEtUgR) [inline src/lib/FlightTasks/tasks/AutoMapper/FlightTaskAutoMapper.cpp]: Is it necessary/desired to publish these all the time even if there is no obstacle avoidance enabled or present?
Could it be conditioned by `COM_OBS_AVOID` either here or inside the `updateAvoidanceDesiredSetpoints()` function?

**Comment by mrivi** [comment]
@MaEtUgR do you mind reviewing again. There are a few changes:
- Send velocity and position setpoints from AutoMapper
- Do not overwrite yaw setpoints with external yaw handling is enabled (necessary for fixed wing)
- fix bug in the check for mission progression when OA is enabled. We need to progress the mission if we are within the xy acceptance radius but either above or below the wp in z. However this condition is always valid for takeoff thus updating to the successive waypoint too soon (takeoff not yet completed)

File: src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp [hunk 2]
```diff
@@ -121,17 +121,22 @@ void ObstacleAvoidance::injectAvoidanceSetpoints(Vector3f &pos_sp, Vector3f &vel
 
 	pos_sp = _sub_vehicle_trajectory_waypoint->get().waypoints[vehicle_trajectory_waypoint_s::POINT_0].position;
 	vel_sp = _sub_vehicle_trajectory_waypoint->get().waypoints[vehicle_trajectory_waypoint_s::POINT_0].velocity;
-	yaw_sp =  _sub_vehicle_trajectory_waypoint->get().waypoints[vehicle_trajectory_waypoint_s::POINT_0].yaw;
-	yaw_speed_sp = _sub_vehicle_trajectory_waypoint->get().waypoints[vehicle_trajectory_waypoint_s::POINT_0].yaw_speed;
+
+	if (!_ext_yaw_active) {
```
Comment (MaEtUgR) [inline src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp]: does obstacle avoidance still work without yaw control or is that just a workaround to test the landing feature for the VTOL?

Comment (mrivi) [reply src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp]: if you have 360 camera coverage, the avoidance doens't need to control yaw. 
Now it's a workaround to test some stuff with vtol

File: src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp [hunk 3]
```diff
@@ -121,17 +121,22 @@ void ObstacleAvoidance::injectAvoidanceSetpoints(Vector3f &pos_sp, Vector3f &vel
...
+		// inject yaw setpoints only if weathervane isn't active
```
[the remark this answers is not in the document]
Comment (mrivi) [reply src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp]: ok

File: src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.hpp [hunk 4]
```diff
@@ -96,18 +98,18 @@ class ObstacleAvoidance : public ModuleParams
 	 * @param prev_wp, previous position triplet
 	 * @param target_acceptance_radius, current position triplet xy acceptance radius
 	 * @param closest_pt, closest point to the vehicle on the line previous-current position triplet
+	 * @param
```
[the remark this answers is not in the document]
Comment (MaEtUgR) [reply src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.hpp]: https://github.com/PX4/Firmware/pull/12009/commits/048d2f548d4073e7c1b52e1faa55eefb57fd2900

**Pull request merged**